vnext/Microsoft.ReactNative/ABIViewManager.h

```diff
@@ -19,32 +19,36 @@
 
 namespace winrt::Microsoft::ReactNative {
 
-class ABIViewManager : public react::uwp::FrameworkElementViewManager {
-  using Super = react::uwp::FrameworkElementViewManager;
+class ABIViewManager : public ::Microsoft::ReactNative::FrameworkElementViewManager {
+  using Super = ::Microsoft::ReactNative::FrameworkElementViewManager;
 
  public:
   ABIViewManager(
       Mso::CntPtr<Mso::React::IReactContext> const &reactContext,
       ReactNative::IViewManager const &viewManager);
 
-  const char *GetName() const override;
+  const wchar_t *GetName() const override;
 
-  folly::dynamic GetExportedViewConstants() const override;
+  void GetExportedViewConstants(const winrt::Microsoft::ReactNative::IJSValueWriter &writer) const override;
 
-  folly::dynamic GetNativeProps() const override;
+  void GetNativeProps(const winrt::Microsoft::ReactNative::IJSValueWriter &writer) const override;
 
-  void UpdateProperties(react::uwp::ShadowNodeBase *nodeToUpdate, const folly::dynamic &reactDiffMap) override;
+  void UpdateProperties(
+      ::Microsoft::ReactNative::ShadowNodeBase *nodeToUpdate,
+      winrt::Microsoft::ReactNative::JSValueObject &props) override;
 
-  folly::dynamic GetCommands() const override;
+  void GetCommands(const winrt::Microsoft::ReactNative::IJSValueWriter &writer) const override;
 
   void DispatchCommand(
       const xaml::DependencyObject &viewToUpdate,
       const std::string &commandId,
-      const folly::dynamic &commandArgs) override;
+      winrt::Microsoft::ReactNative::JSValueArray &&commandArgs) override;
```

Comment:
>JSValueArray [](start = 37, length = 12)

It would be better if we change the use of the JSValueArray to IJSValueReader. Using of JSValue assumes that we always convert value to JSValue that often causes extra memory allocations. While the IJSValueReader avoids the extra memory allocations when passing parameters.

Reply:
Agreed... for the future.